src/engine/runtime.js

```diff
@@ -587,7 +597,8 @@ class Runtime extends EventEmitter {
                 // give up if any threads with the top block are running.
                 for (let j = 0; j < instance.threads.length; j++) {
                     if (instance.threads[j].topBlock === topBlockId &&
-                        instance.threads[j].target === target) {
+                        instance.threads[j].target === target &&
+                        !instance.threads[j].status === Thread.STATUS_DONE) {
```

Comment:
3. Start hats aren't added every other frame because we keep them around as "done" threads